### PR TITLE
feat: `keypair_name` and `last_used_derivation_index` columns added to `accounts` table

### PR DESCRIPTION
**Pull request by saledjenic**
- `keypair_name` added to `accounts` table, all accounts derived from the same master key have the same keypair name and also no two keypairs share the same keypair name (keypair name is unique per keypair)
- `last_used_derivation_index` added to `accounts` table, cause we need to maintain the highest index been used for the derivations made within the same keypair

@status-im/mobile-devs I am not sure if you're using `latest_derived_path` from the `settings` table or not (the only thing why the desktop app used it for was to properly create/derive next wallet account), but if not I would remove that column form the `settings` table within this PR?

### CODE REVIEW

**Comment by ilmotta** [comment]
> 

Hey @saledjenic, I searched the mobile codebase, but I couldn't find any reference to `latest_derived_path`. It should be fine to remove the column, but maybe somebody else has a concern.

**Comment by qfrank** [comment]
> I am not sure if you're using latest_derived_path from the settings table  

Hi @saledjenic , mobile used `latest_derived_path`

**Review by richard-ramos — APPROVED** [review]

**Review by qfrank — APPROVED** [review]

**Comment by saledjenic** [comment]
@richard-ramos @qfrank @alaibe please review this one again cause I added second and third commit as part of this PR, cause I need those additional changes for my "add wallet account" desktop related work.

- first commit: `keypair_name and last_used_derivation_index columns added to accounts table`, this one was already reviewed
- second commit `create acc from private key endpoint added` a new function `CreateAccountFromPrivateKey` was added, which creates an account from the provided private key without creating keystore files and adding anything to db.
- third commit ` accounts and wallet api sorted out` api endpoints are sorted out a bit, those unused are removed some new are added according to the desktop app needs.
  - **removed endpoints (accounts api)**:
    - `AddAccountWatch`
    - `AddAccountWithMnemonic`
    - `AddAccountWithMnemonicPasswordVerified`
    - `AddAccountWithMnemonicAndPath`
    - `AddAccountWithMnemonicAndPathPasswordVerified`
    - `AddAccountWithPrivateKeyPasswordVerified`
    - `AddAccountWithPrivateKey`
    - `GenerateAccount`
    - `GenerateAccountPasswordVerified`
    - `GenerateAccountWithDerivedPath`
  - **newly added endpoints (accounts api)**:
    - `AddAccount`
    - `ImportPrivateKey`
    - `ImportMnemonic`
    - `GetRandomMnemonic`
  - **removed endpoints (wallet api)**:
    - `GetDerivedAddressForPath`
    - `GetDerivedAddressesForPath`
    - `GetDerivedAddressesForMnemonicWithPath`
    - `GetDerivedAddressForPrivateKey`
    - `GetDerivedAddressDetails`
  - **newly added endpoints (wallet api)**:
    - `GetDerivedAddresses`
    - `GetDerivedAddressesForMnemonic`
    - `GetAddressDetails`

File: multiaccounts/accounts/database.go [hunk 1]
```diff
@@ -129,6 +144,112 @@ func (db *Database) Close() error {
...
+		WHERE 
+			key_uid = ? 
+		AND
+			NOT chat`,
+		keypairName, index, keyUID)
+
+	return err
+}
+
+func updateAccountsTableIfNeeded(db *sql.DB) error {
```
Comment (alaibe) [inline multiaccounts/accounts/database.go]: nit: this function name could probably have a better name e.g what to update and what is needed?

**Review by alaibe — APPROVED** [review]

File: multiaccounts/accounts/database.go (hunk 1 above)
Comment (saledjenic) [reply multiaccounts/accounts/database.go]: fully agree, will update it later

Comment (saledjenic) [reply multiaccounts/accounts/database.go]: done

**Review by qfrank — APPROVED** [review]

**Pull request merged**